### ISSUE TITLE
JSON Output

### ISSUE THREAD

**Issue by wjdp** [newtheatre/nt-tickets#8]
We currently integrate with the main site by outputing bare HTML. For the new site, if it'll be static, can't use this — https://github.com/newtheatre/website/issues/7.

Implement a JSON endpoint, can have a single for the sidebar and what'son use case and control what is returned with URL params.

Something like

`/api/1/shows?count=1&seasons=main,fringe` for sidebar
`/api/1/shows?count=999&seasons=main,fringe` for what's on page

**Comment by harry-bridge** [comment]
Do we want an api call for booking shows like

`/api/book/1` that returns show details like `{"results":[{"tickets_left":20, "show_times":[{"1800", "1900"}]]}`

**Comment by wjdp** [comment]
GET all shows
GET show by id

``` json
{
 "name": "show name",
 "occurrences": [
   {
    "id": 999,
    "datetime": 12345678,
    "tickets_left": 10 
   } 
 ]
}
```

POST for ticket booking, returns success or fail

User sends: 
- occurrence id
- no of tickets
- name
- email

**Comment by tomtom5152** [comment]
What you want here gentleman is Django Rest Framework

**Comment by harry-bridge** [comment]
Funny that, this is what I'm using

**Comment by davidtaylorhq** [comment]
This API could also be used for the foyer screens, whenever someone gets around to making a web page for them to display

**Comment by harry-bridge** [comment]
This is true, can we display a web page on them now? Could be persuaded to make it tonight

**Comment by davidtaylorhq** [comment]
@pickleshb Well we have wifi login, and we have raspberry pis with wifi. I can definitely make a raspberry pi connect to eduroam & display a webpage on boot

**Comment by davidtaylorhq** [comment]
So we have all the parts, just need to stick them together

**Comment by wjdp** [comment]
We need to have a chat about which data source will be used. The ticketing system will have basic information and ticket availability, and the website will have more comprehensive show information.

**Comment by harry-bridge** [comment]
is there some way of getting an output from jekyll api style?

**Comment by davidtaylorhq** [comment]
@pickleshb So what we want is a page on the website specifically designed for the screens - hidden from robots and without any navigation to it

**Comment by harry-bridge** [comment]
Well hidden from google's robots

**Comment by harry-bridge** [comment]
Do we know what size the screens are?

**Comment by davidtaylorhq** [comment]
@pickleshb Hidden from all robots using deny:all

**Comment by wjdp** [comment]
> Harry Bridge So what we want is a page on the website specifically designed for the screens - hidden from robots and without any navigation to it

yeep

@pickleshb see 
- https://history.newtheatre.org.uk/feeds/people.json
- https://history.newtheatre.org.uk/feeds/smug_albums.json

**Comment by harry-bridge** [comment]
But that can be ignored :P

**Comment by davidtaylorhq** [comment]
Well it's not showing any private information, so if people really want to index it, go ahead.

I suggest making the page assuming that the screens are 1080p

**Comment by samozzy** [comment]
This thread is old, but did any progress get made? The new Jekyll site is taking shape so it would be great to get this functionality.

**Comment by harry-bridge** [comment]
@samozzy Tell me what you want and in what format and I’ll make it a thing for you

**Comment by samozzy** [comment]
- List of all shows on the homepage 
- Ability to book tickets (unsure how this works) 
JSON makes sense as a format, but again I'm not too sure how to get that to Jekyll. I have got a prototype page that generates the list from a local JSON file so once Jekyll can see the external file it shouldn't be too tricky to finish the job.

**Comment by harry-bridge** [comment]
That sounds rather trivial

I'll have a think about the easiest way to book a show but it'll be something like `{'show_pk', 'occurrence_pk', 'num_seats', 'name', 'email' 'api_key'}` with a success or fail response 

To parse json there are a load of methods in jQuery JSON.parse() comes to mind, but there are probably other ways, equally there is a .getJSON() method which is a nice shorthand get request for this use
